pages/docs/tutorials/generate-code.md

```diff
@@ -0,0 +1,136 @@
+---
+title: Generate code 
+description: In this tutorial, you'll learn how to generate code from your AsyncAPI document.
+weight: 130
+---
+
+## Introduction
+
+In this tutorial, you'll learn how to generate code from your AsyncAPI document using the AsyncAPI generator tool.
+
+## Background context
+The [AsyncAPI Generator](https://github.com/asyncapi/generator) is a tool that you can use to generate whatever you want based on the AsyncAPI document. You can generate docs and code. It can be used as a library in Nodejs application or through the [AsyncAPI CLI](https://github.com/asyncapi/cli).
+
+The generator tool supports a number of templates to generate code for a variety of different languages and protocols as the output. These templates help to specify what exactly must be generated, and in this tutorial, you use [Nodejs template](https://github.com/asyncapi/nodejs-template).
+
+## Installation guide
```

Comment:
we need more than just node/npm instructions, I think I already pointed you to https://github.com/asyncapi/generator/pull/818/files#diff-6c85f832c3eae8efc6830b5e9e56020250a891fa7fd58498a8506c4600ab3c34R32-R53 but I'm might mix PRs 😄 

bigger question related to it and convo from https://asyncapi.slack.com/archives/C02QY9FMM18/p1668535611547909

cc @alequetzalli 

since tutorials should also be used as standalone, then it means `CLI installation` instructions will show up in more than one tutorial.

I strongly suggest we start reusing content. I mean having one file where this CLI installation is described, and then we include it in as many other documents as needed.

It is technically possible as we already had a convo about it with @magicmatatjahu in this PR https://github.com/asyncapi/website/pull/886#issuecomment-1222760522

I also tested it now quickly on local.
tl;dr it is possible out of the box

VSCode 👇🏼 

![Screenshot 2022-11-16 at 13 28 35](https://user-images.githubusercontent.com/6995927/202180896-5e25a2e0-b1e5-4740-8421-a50ca9109f3f.png)

Result in UI 👇🏼 

![Screenshot 2022-11-16 at 13 29 22](https://user-images.githubusercontent.com/6995927/202181044-7aa8ee3e-9087-4db1-9774-42be76a05d1d.png)

Reply:
Remember that we shouldn't write fragments/reusable content in the `pages/**` folder, because then NextJS will treat (as in your example) the `pages/docs/fragments/**` as separate pages in website. We should move them in the `{root}/docs` folder or something like.

cc @derberg @alequetzalli

Reply:
**Cool! I love the idea of having these reusable components!**

1) I don't think we need to mention the Generator Tool tho? That seemed like an example only relevant to that one tutorial...
2) Are you volunteering to create this? 😂 I am asking just so I follow as to when you hope to start implementing this. Is this something that can be easily/quickly done before this month ends with GSoD program? 🤔

Reply:
nvm... resolved via https://github.com/asyncapi/website/pull/1113

Reply:
> we need more than just node/npm instructions, I think I already pointed you to https://github.com/asyncapi/generator/pull/818/files#diff-6c85f832c3eae8efc6830b5e9e56020250a891fa7fd58498a8506c4600ab3c34R32-R53 but I'm might mix PRs 😄

Hi @derberg, 
In having more than the node/npm instructions, do you mean adding installation instructions for other operating systems as shown in the link you pointed above.

I remember you once pointed sticking to one OS here 
<img width="558" alt="convo" src="https://user-images.githubusercontent.com/52763841/202599055-552efe62-beb6-4c08-85b7-9a39475a38ef.PNG">

Reply:
Got it!

Reply:
"To generate code, you need to install the AsyncAPI CLI that allows you to generate code..."